### PR TITLE
ui: implement basic keyboard shortcuts

### PR DESCRIPTION
**Pull request by zannkukai**
Adds keyboard shortcuts to the professional interface. When the focus
is not set into an input element, if user press on some keyboard key
(or keys combination) an action is done (redirection, set focus, ...).

Co-Authored-by: Renaud Michotte <renaud.michotte@gmail.com>

## How to test?

`npm ci`

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?

### CODE REVIEW

**Review by iGormilhit — APPROVED** [review]
Commit message approved.

**Review by benerken — APPROVED** [review]

File: projects/admin/src/app/widgets/custom-shortcut-help/custom-shortcut-help.component.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+@Component({
+  selector: 'admin-custom-shortcut-help',
+  templateUrl: './custom-shortcut-help.component.html',
+  styleUrls: ['./custom-shortcut-help.component.scss']
+})
+export class CustomShortcutHelpComponent {
+
+  @Input() title = this._translateService.instant('Available Shortcuts');
+  hotkeys = this._hotKeysService.getShortcuts();
```
Comment (sebastiendeleze) [inline projects/admin/src/app/widgets/custom-shortcut-help/custom-shortcut-help.component.ts:32]: Some comments can be added for properties

File: projects/admin/src/app/circulation/item/item.component.html [hunk 2]
```diff
@@ -57,7 +57,7 @@
     <a [routerLink]="['/records','documents','detail', item.document.pid]" *ngIf="item.document.title | mainTitle as title ">
       {{ isCollapsed ? (title | truncateText: 12) : title }}
     </a>
-    <ng-container *ngIf="document.contribution && document.contribution.length > 0">
+    <ng-container *ngIf="document && document.contribution && document.contribution.length > 0">
```
Comment (sebastiendeleze) [inline projects/admin/src/app/circulation/item/item.component.html:60]: Could be added to the commit message

File: projects/admin/src/app/circulation/patron/main/main.component.ts [hunk 3]
```diff
@@ -41,26 +90,42 @@ export class MainComponent implements OnInit, OnDestroy {
   /** Subsription to current patron */
   private _patronSubscription$: Subscription;
 
+  /** Get current organisation
+   *  @return: current organisation
```
Comment (sebastiendeleze) [inline projects/admin/src/app/circulation/patron/main/main.component.ts:94]: should be `@returns` and without `:`

File: projects/admin/src/app/circulation/patron/main/main.component.ts [hunk 4]
```diff
@@ -92,13 +154,31 @@ export class MainComponent implements OnInit, OnDestroy {
     this._patronService.clearPatron();
   }
 
-  /** Get current organisation
-   *  @return: current organisation
-   */
-  get organisation() {
-    return this._organisationService.organisation;
+  clearPatron() {
```
Comment (sebastiendeleze) [inline projects/admin/src/app/circulation/patron/main/main.component.ts:157]: Doc block is missing

**Review by sebastiendeleze — APPROVED** [review]
Nice feature 🎉

**Review by lauren-d — APPROVED** [review]

**Pull request merged**